ecommerce-frontend/src/user/AdminDashboard.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import Layout from "../core/Layout";
+import { isAuthenticated } from "../auth";
+import { Link } from "react-router-dom";
+
+const AdminDashboard = () => {
+  const {
+    user: { _id, name, email, role },
+  } = isAuthenticated();
```

Comment:
Don't check `isAuthenticated` at every stage. Check it once and then save its value and reuse it. You can use. `localstorage`